Automattic/vip-go-mu-plugins
# Search: query ES for the last post ID in get_last_post_id()

## Description
`get_last_post_id()` only queries the DB for the last post ID, which means if we have an extra post in the ES index, our `validate-contents` will not include it in the range and therefore, it will be missed.

## Changelog Description
### Plugin Updated: Enterprise Search

Fix bug where `get_last_post_id()` doesn't account for extra ID in ES index.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
0. Check out PR
1. Create a post - `vip dev-env --slug test exec -- wp post create --post_title=test --post_status=publish`
2. Delete the created post directly in DB - `vip dev-env --slug test exec -- wp db cli + DELETE FROM wp_posts WHERE ID = 50;`
3. Flush the cache - `vip dev-env --slug test exec -- wp cache flush`
4. Run `dev-env exec --slug test -- wp vip-search health validate-contents`

You should see the last post in ES in the range. For this example, we are using `50`:

```
Validating posts 1 - 50...✘ (attempted to reconcile)
```
